src/core/public/application/utils/get_app_info.ts

```diff
@@ -28,29 +28,27 @@ export function getAppInfo(app: App): PublicAppInfo {
     status: app.status!,
     navLinkStatus,
     appRoute: app.appRoute!,
-    meta: {
-      keywords: app.meta?.keywords ?? [],
-      searchDeepLinks: getSearchDeepLinkInfos(app, app.meta?.searchDeepLinks),
-    },
+    keywords: app.keywords ?? [],
+    deepLinks: getDeepLinkInfos(app.deepLinks),
   };
 }
 
-function getSearchDeepLinkInfos(
-  app: App,
-  searchDeepLinks?: AppSearchDeepLink[]
-): PublicAppSearchDeepLinkInfo[] {
-  if (!searchDeepLinks) {
-    return [];
-  }
+function getDeepLinkInfos(deepLinks?: AppDeepLink[]): PublicAppDeepLinkInfo[] {
+  if (!deepLinks) return [];
 
-  return searchDeepLinks.map(
-    (rawDeepLink): PublicAppSearchDeepLinkInfo => {
+  return deepLinks.map(
+    (rawDeepLink): PublicAppDeepLinkInfo => {
+      const navLinkStatus =
+        rawDeepLink.navLinkStatus === AppNavLinkStatus.default
+          ? AppNavLinkStatus.hidden
+          : rawDeepLink.navLinkStatus!;
```

Comment:
Hi @joshdover, I have one question here. 

Checking the conditionals to default the main `navLinkStatus` at line 19, it is possible to have an `app.status: "unaccessible"` and `app.navLinkStatus: "visible"` at the same time, I assumed this is an intentional situation.

Just to confirm, as I did not prevent this situation in the nested `navLinkStatus` inside `deepLinks` either, to follow the same criteria.

Reply:
Discussed this sync, we're going to keep this consistent for now. If needed, this can be fixed in a follow up outside this change.

I think one aspect here that is not working quite right is the lack of a default option. `getAppInfo` should always return an object where there is a `navLinkStatus` for each (top-level) deep link, however right now it doesn't. I think we can fix this by populating the default in the same place we populate it for the top-level `navLinkStatus`: https://github.com/elastic/kibana/blob/master/src/core/public/application/application_service.tsx#L168